upup/pkg/fi/dockertasks/copydockerimage.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockertasks
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"k8s.io/kops/upup/pkg/fi"
+)
+
+// CopyDockerImage copies a docker image from a source registry, to a target registry,
+// typically used for highly secure clusters.
+//go:generate fitask -type=CopyDockerImage
+type CopyDockerImage struct {
+	Name        *string
+	SourceImage *string
+	TargetImage *string
+}
+
+var _ fi.CompareWithID = &CopyDockerImage{}
+
+func (e *CopyDockerImage) CompareWithID() *string {
+	return e.Name
+}
+
+func (e *CopyDockerImage) Find(c *fi.Context) (*CopyDockerImage, error) {
```

Comment:
This needs to be implemented because the chance of clobbering changed images that already exist in the repository.  Filing an issue to get this done.